accommodation/views.py

```diff
@@ -94,7 +83,59 @@ def get(self, request, ):
                 'review'      : accommodation.count,
                 'price'       : accommodation.price,
                 'url'         : accommodation.accommodationimage_set.first().image_url,
-            } for accommodation in accommodations.filter(**filter_set).order_by(ordering_options[ordering])
+            } for accommodation in accommodations.order_by(ordering_options[ordering]) 
         ]
 
         return JsonResponse({'data': data}, status=200)
+
+class AccommodationDetailView(View):
+    def get(self, request, accommodation_id):
+        try:
+            accommodation = Accommodation.objects.get(id=accommodation_id)
+            review_count  = Review.objects.filter(accommodation=accommodation_id).count()
+            
+            data = {
+                'id'            : accommodation.id,
+                'name'          : accommodation.name,
+                'description'   : accommodation.description,
+                'total_rate'    : accommodation.rate,
+                'review_number' : review_count,
+                'room'          : [{
+                    'name'           : room.name,
+                    'option'         : [
+                        {
+                            'name' : option.name
+                        } for option in room.option.all()
+                    ],
+                    'price'          : room.price,
+                    'basic_capacity' : room.basic_capacity,
+                    'max_capacity'   : room.maximum_capacity,
+                    'extra_price'    : room.extra_price,
+                    'image'          : [
+                        {
+                            'image_url' : image.image_url 
+                        } for image in room.accommodation.accommodationimage_set.all()
+                    ],
+                } for room in accommodation.room_set.all()],
+                'main_image'    : accommodation.accommodationimage_set.first().image_url,
```

Comment:
```suggestion
                'main_image'    : accommodation.accommodationimage_set.first().image_url,
```
first() 장고에 있는 좋은 메소드를 사용했다는 측면에서는 좋은 코드이나, 조금더 확장해서 성능적인 측면에 대해서 생각해보자면, 
all()[0]이 더 나을 수 있습니다. 
왜 그런지 확인해보고 답해주세요.

Reply:
넵! 공부하고 다시 코멘트 남기겠습니다!

Reply:
아주 소박한 추측 | first() 메소드는 아무것도 안들어오면 None을 반환하지만... all()[0]은 IndexError를 반환하기 때문이라고 생각합니다..!!!!!

Reply:
네 현재상황에서는 first()가 큰 영향이 없는데, 캐싱된 데이터를 순회할 경우에는 first() 사용시 DB hit가 더 발생하고,  all()[0] 시에는 발생되지 않습니다. 추후에 적용 바랍니다. 현재 get은 근데 아에 데이터 캐싱이 없었네요 ㅋㅋ

Reply:
앗 알려주셔서 감사합니다 지훈님!!